lib/src/values/random_logic_value.dart

```diff
@@ -0,0 +1,79 @@
+/// Copyright (C) 2021-2022 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// random_logic_value.dart
+/// Random Logic Value generation extension.
+///
+/// 2023 May 18
+/// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+///
+
+part of values;
+
+/// Allows random generation of [LogicValue] for [BigInt] and [int].
+extension RandLogicValue on math.Random {
+  /// Generate unsigned random [BigInt] value that consists of
+  /// [numBits] bits.
+  BigInt _nextBigInt({required int numBits}) {
+    var result = BigInt.zero;
+    for (var i = 0; i < numBits; i += 32) {
+      result = (result << 32) | BigInt.from(nextInt(1 << 32));
+    }
+    return result & ((BigInt.one << numBits) - BigInt.one);
+  }
+
+  /// Generate unsigned random [LogicValue] based on [width] and [max] num.
+  /// The random number can be mixed in invalid bits x and z by set
+  /// [includeInvalidBits] to `true`. [max] only work when [includeInvalidBits]
+  /// is set to false else an exception will be thrown.
+  LogicValue nextLogicValue({
+    required int width,
+    LogicValue? max,
```

Comment:
It might make more sense to make `max` a `dynamic` that supports either `int` or `BigInt`?  In which case, need to check that it's not negative.  Otherwise, if we keep `LogicValue`, we should check that it's valid.